framework/src/base/FEProblemBase.C

```diff
@@ -134,6 +134,10 @@ validParams<FEProblemBase>()
                         false,
                         "Do not explicitly store zero values in "
                         "the Jacobian matrix if true");
+  params.addParam<bool>("update_jacobian_preallocation",
+                        false,
+                        "Update Jacobian preallocation during each nonlinear iteration"
+                        "if true");
```

Comment:
Can there be another option to update preallocation at every time-step instead of each nonlinear iteration? I mean can the user decide when to update preallocation depending on the problem? I probably will not need it at every iteration.